src/System.Text.Json/src/System/Text/Json/Serialization/ReadStackFrame.cs

```diff
@@ -41,35 +44,56 @@ internal struct ReadStackFrame
         public int PropertyIndex;
         public List<PropertyRef> PropertyRefCache;
 
-        // The current JSON data for a property does not match a given POCO, so ignore the property (recursively).
-        public bool Drain;
+        public bool IsProcessingCollectionForClass()
+        {
+            return IsProcessing(ClassType.Enumerable | ClassType.Dictionary | ClassType.IDictionaryConstructible);
+        }
 
-        public bool IsCollectionForClass => IsEnumerable || IsDictionary || IsIDictionaryConstructible;
-        public bool IsCollectionForProperty => IsEnumerableProperty || IsDictionaryProperty || IsIDictionaryConstructibleProperty;
+        public bool IsProcessingCollectionForProperty()
```

Comment:
A lot of these checks are doing redundant work (at least where they are called), so I am wondering if a cleaner solution would be to make those checks less redundant rather than turning the enums into a bit flag (or maybe we do both).

Reply:
I'm open for other ways to do this, but using bit flags was the fastest and cleanest solution I came up with.